src/compiler/html/inline-esm-import.ts

```diff
@@ -25,7 +25,7 @@ export async function optimizeEsmImport(config: d.Config, compilerCtx: d.Compile
 
   // If the script is too big, instead of inlining, we hash the file and change
   // the <script> to the new location
-  if (content.length > MAX_JS_INLINE_SIZE) {
+  if (config.allowInlineScripts && content.length > MAX_JS_INLINE_SIZE) {
```

Comment:
Another possible solution: allow a config value for `MAX_JS_INLINE_SIZE` so can be set it to `0` to prevent inlining